path_roles.go

```diff
@@ -208,6 +218,53 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		role.PermanentlyDelete = false
 	}
 
+	// update and validate SignInAudience if provided
+	if signInAudience, ok := d.GetOk("sign_in_audience"); ok {
+		signInAudienceValue, ok := signInAudience.(string)
+		if !ok {
+			return logical.ErrorResponse("Invalid type for sign_in_audience field. Expected string."), nil
+		}
+
+		validSignInAudiences := []string{"AzureADMyOrg", "AzureADMultipleOrgs", "AzureADandPersonalMicrosoftAccount", "PersonalMicrosoftAccount"}
+		valid := false
+		for _, validValue := range validSignInAudiences {
+			if signInAudienceValue == validValue {
+				valid = true
+				break
+			}
+		}
+		if !valid {
+			validValuesString := strings.Join(validSignInAudiences, ", ")
+			return logical.ErrorResponse("Invalid value for sign_in_audience field. Valid values are: %s", validValuesString), nil
+		}
```

Comment:
The Vault SDK `strutil` package has some nice helpers for this type of validation. Might consider simplifying to something like:
```go
validSignInAudiences := []string{"AzureADMyOrg", "AzureADMultipleOrgs", "AzureADandPersonalMicrosoftAccount", "PersonalMicrosoftAccount"}
if !strutil.StrListContains(validSignInAudiences, signInAudienceValue) {
	validValuesString := strings.Join(validSignInAudiences, ", ")
	return logical.ErrorResponse("Invalid value for sign_in_audience field. Valid values are: %s", validValuesString), nil
}
```

Reply:
Thank you! Thank looks a lot nicer, updated in the last commit!